lib/certs.go

```diff
@@ -0,0 +1,330 @@
+/*-
+ * Copyright 2016 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package lib
+
+import (
+	"bufio"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/binary"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/square/certigo/jceks"
+	"github.com/square/certigo/pkcs7"
+	"golang.org/x/crypto/pkcs12"
+)
+
+const (
+	// nameHeader is the PEM header field for the friendly name/alias of the key in the key store.
+	nameHeader = "friendlyName"
+
+	// fileHeader is the origin file where the key came from (as in file on disk).
+	fileHeader = "originFile"
+)
+
+var fileExtToFormat = map[string]string{
+	".pem":   "PEM",
+	".crt":   "PEM",
+	".p7b":   "PEM",
+	".p7c":   "PEM",
+	".p12":   "PKCS12",
+	".pfx":   "PKCS12",
+	".jceks": "JCEKS",
+	".jks":   "JCEKS", // Only partially supported
+	".der":   "DER",
+}
+
+var badSignatureAlgorithms = [...]x509.SignatureAlgorithm{
+	x509.MD2WithRSA,
+	x509.MD5WithRSA,
+	x509.SHA1WithRSA,
+	x509.DSAWithSHA1,
+	x509.ECDSAWithSHA1,
+}
+
+// ReadPEMFromFiles will read PEM blocks from the given set of inputs.
+func ReadPEMFromFiles(files []*os.File, format string, password func(string) string, callback func(*pem.Block)) error {
```

Comment:
This will actually read any file format we support, so I think it should be clear.  We could call it "ReadAsPEM" instead (and comment on what it does)